platform/jobs/edxPlatformPythonUnitTestsPr.groovy

```diff
@@ -49,63 +26,130 @@ catch (any) {
+                             workerLabel: 'ficus-jenkins-worker',
+                             whitelistBranchRegex: /open-release\/ficus.master/,
+                             context: 'jenkins/ficus/python',
+                             triggerPhrase: 'ficus run python',
+                             defaultTestengBranch: 'origin/open-release/ficus.master'
+                             ]
+
+Map privateFicusJobConfig = [ open: false,
+                              jobName: 'ficus-python-unittests-pr_private',
+                              subsetJob: 'edx-platform-test-subset_private',
+                              repoName: 'edx-platform-private',
+                              coverageJob: 'edx-platform-unit-coverage_private',
+                              workerLabel: 'ficus-jenkins-worker',
+                              whitelistBranchRegex: /open-release\/ficus.master/,
+                              context: 'jenkins/ficus/python',
+                              triggerPhrase: 'ficus run python',
+                              defaultTestengBranch: 'origin/open-release/ficus.master'
+                              ]
+
+List jobConfigs = [ publicJobConfig,
+                    privateJobConfig,
+                    publicGinkgoJobConfig,
+                    privateGinkgoJobConfig,
+                    publicFicusJobConfig,
+                    privateFicusJobConfig
+                    ]
+
+// Iterate over the job configs to create individual build flow jobs
+jobConfigs.each { jobConfig ->
+
+    buildFlowJob(jobConfig.jobName) {
+
+        if (!jobConfig.open.toBoolean()) {
+            authorization GENERAL_PRIVATE_JOB_SECURITY()
         }
         properties {
-              githubProjectUrl(JENKINS_PUBLIC_GITHUB_BASEURL + jobConfig['platformUrl'])
+              githubProjectUrl("https://github.com/edx/${jobConfig.repoName}/")
         }
         logRotator JENKINS_PUBLIC_LOG_ROTATOR()
         concurrentBuild()
         label('flow-worker-python')
         checkoutRetryCount(5)
         environmentVariables {
-            env('SUBSET_JOB', jobConfig['subsetJob'])
-            env('REPO_NAME', jobConfig['repoName'])
-            env('COVERAGE_JOB', jobConfig['coverageJob'])
+            env('SUBSET_JOB', jobConfig.subsetJob)
+            env('REPO_NAME', jobConfig.repoName)
+            env('COVERAGE_JOB', jobConfig.coverageJob)
         }
         parameters {
-            stringParam('WORKER_LABEL', jobConfig['workerLabel'], 'Jenkins worker for running the test subset jobs')
+            stringParam('WORKER_LABEL', jobConfig.workerLabel, 'Jenkins worker for running the test subset jobs')
         }
         multiscm {
             git {
                 remote {
-                    url(JENKINS_PUBLIC_GITHUB_BASEURL + jobConfig['testengUrl'] + '.git')
-                    if (!jobConfig['open'].toBoolean()) {
-                        credentials(jobConfig['testengCredential'])
```

Comment:
doublechecking my understanding of this change - is it that jenkins workers don't need credentials for cloning this repo because it's open?

Reply:
In the past, we had considered creating a private version of testeng-ci, but the reasons to do so never fully materialized